apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx

```diff
@@ -147,10 +157,18 @@ export function ImportLedgerAccounts() {
                     variant="primary"
                     before={<UnlockedLockIcon />}
                     text="Unlock"
-                    onClick={() => {
-                        // TODO: Do work to actually import the selected accounts once we have
-                        // the account infrastructure setup to support Ledger accounts
-                        navigate(accountsUrl);
+                    loading={importLedgerAccountsMutation.isLoading}
+                    onClick={async () => {
+                        try {
+                            await importLedgerAccountsMutation.mutateAsync(
+                                selectedLedgerAccounts
+                            );
+                            navigate(accountsUrl);
+                        } catch (error) {
+                            toast.error(
```

Comment:
Similarly could use the onError of the mutation. I'm a little indifferent just wanted to flag.